cpp/include/rapidsmpf/streaming/core/channel.hpp

```diff
@@ -84,8 +86,11 @@ class Channel {
     [[nodiscard]] bool empty() const noexcept;
 
   private:
-    Channel() = default;
-    coro::ring_buffer<Message, 1> rb_;
+    Channel(std::shared_ptr<SpillableMessages> spillable_messages)
+        : sm_{std::move(spillable_messages)} {}
+
+    coro::ring_buffer<SpillableMessages::MessageId, 1> rb_;
```

Comment:
I didn't spot this before, but a `MessageId` is "just" a `uint64`. Shall we (probably in a followup) make it a strong type:

```
enum class MessageId : uint64_t {}
```

And don't allow the user to conjure a message id from nowhere.